streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionResponseWrapperSerde.java

```diff
@@ -62,6 +71,42 @@ public void setIfUnset(final SerdeGetter getter) {
             }
         }
 
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            this.upgradeFromV0 = upgradeFromV0(configs);
+        }
+
+        private static boolean upgradeFromV0(final Map<String, ?> configs) {
+            final Object upgradeFrom = configs.get(StreamsConfig.UPGRADE_FROM_CONFIG);
+            if (!(upgradeFrom instanceof String)) {
```

Comment:
We don't need this check. `StreamsConfig` ensures that it will be a `String`. Cf `StreamConfig`:
```
.define(UPGRADE_FROM_CONFIG,
        ConfigDef.Type.STRING,
```

Reply:
Replaced with a not-null check in 8ebbaaf38ee79642ccb322694de7556f260f93ca.